src/Microsoft.AspNetCore.Localization/CookieRequestCultureProvider.cs

```diff
@@ -118,5 +145,8 @@ public static ProviderCultureResult ParseCookieValue(string value)
 
             return new ProviderCultureResult(cultureName, uiCultureName);
         }
+
+        private static bool IsValidCulture(string cultureName) =>
+            CultureInfo.GetCultures(CultureTypes.SpecificCultures).Any(c => c.Name == cultureName);
```

Comment:
> IMO, we could just move the test to the actual middleware and have it complain if we returned a `ProviderCultureResult` which we would not use.

@pranavkm If I move the check to the `RequestLocalizationMiddleware` I have two points to discuss: 
1. `_options` field need to be `static` - which is the thing that you don't like 😄 - to access the `SupportedCultures`
2. `GetCultureInfo()` accept the supported cultures, in our case we need to check both of `SupportedCultures` & `SupportedUICultures`